src/components/Admin.js

```diff
@@ -54,9 +55,13 @@ const Admin = () =>
                         path="/admin/users"
                         exact
                         component={Users}/>
+                    <Route
+                        path="/admin/services/:serviceName"
```

Comment:
This is rendering the name into the id field like it would from App.js. So you're ending up using the findServiceById with the service name as a parameter. Is querying by name a requirement? If so, you might need to add some more logic to handle both the id search and the name search.